pkg/cmd/workspace/create.go

```diff
@@ -199,11 +233,44 @@ func init() {
 	CreateCmd.Flags().StringVar(&providerFlag, "provider", "", "Specify the provider (e.g. 'docker-provider')")
 	CreateCmd.Flags().StringVarP(&ideFlag, "ide", "i", "", "Specify the IDE ('vscode' or 'browser')")
 	CreateCmd.Flags().StringVarP(&targetNameFlag, "target", "t", "", "Specify the target (e.g. 'local')")
+	CreateCmd.Flags().StringVar(&customImageFlag, "custom-image", "", "Create the project with the custom image passed as the flag value; Requires setting --custom-image-user flag as well")
+	CreateCmd.Flags().StringVar(&customImageUserFlag, "custom-image-user", "", "Create the project with the custom image user passed as the flag value; Requires setting --custom-image flag as well")
+	CreateCmd.Flags().StringVar(&devcontainerPathFlag, "devcontainer-path", "", "Automatically assign the devcontainer builder with the path passed as the flag value")
+	CreateCmd.Flags().StringVar(&builderFlag, "builder", "", fmt.Sprintf("Specify the builder (currently %s/%s/%s)", string(create.AUTOMATIC), string(create.DEVCONTAINER), string(create.NONE)))
+
 	CreateCmd.Flags().BoolVar(&manualFlag, "manual", false, "Manually enter the git repositories")
 	CreateCmd.Flags().BoolVar(&multiProjectFlag, "multi-project", false, "Workspace with multiple projects/repos")
 	CreateCmd.Flags().BoolVarP(&codeFlag, "code", "c", false, "Open the workspace in the IDE after workspace creation")
 }
 
+func validateFlags() error {
+	if multiProjectFlag && (builderFlag != "" || devcontainerPathFlag != "" || customImageFlag != "" || customImageUserFlag != "") {
+		return errors.New("Can't use single project only flags (--builder, --devcontainer, --custom-image, --custom-image-user) with --multi-project flag.")
+	}
+
+	if devcontainerPathFlag != "" && (customImageFlag != "" || customImageUserFlag != "") {
+		return errors.New("Can't declare devcontainer file path with custom image and user flags. Choose either.")
+	}
+
+	if builderFlag != "" && builderFlag != string(create.AUTOMATIC) && builderFlag != string(create.DEVCONTAINER) && builderFlag != string(create.NONE) {
```

Comment:
Would making `builderFlag` an enum make this easier to maintain once more build types start coming in? I haven't found a drop-in solution, [this](https://stackoverflow.com/a/70541824) looks like it could work